proof_of_concept/definitions/asset_id.py

```diff
@@ -0,0 +1,82 @@
+"""Identifier for assets."""
+from typing import Any, cast, Type
+
+
+class AssetId(str):
```

Comment:
Nice, I like how you implemented this :)

Reply:
Actually, getting subclassing str to work with mypy is still a bit of a mystery. I'm getting an error message that I shouldn't be getting I think, so I'm going to have to see if this is a known bug... but the `#type: ignore` works for now.

Reply:
Check.